networkit/cpp/graph/GraphTools.cpp

```diff
@@ -74,6 +74,45 @@ node randomNode(const Graph &G) {
     return v;
 }
 
+std::vector<node> randomNodes(const Graph &G, count n) {
+    std::vector<node> selectedNodes;
+    std::vector<bool> alreadySelected(G.numberOfNodes(), false);
+
+    if (n >= G.numberOfNodes()) {
+        for (const auto sample : G.nodeRange()) {
+            selectedNodes.push_back(sample);
+        }
+    } else if (n > G.numberOfNodes() / 2) { // in order to minimize the calls to randomNode
+                                            // we randomize the ones that aren't pivot
+                                            // if the are more to be selected than not-selected
+        std::fill(alreadySelected.begin(), alreadySelected.end(), true);
+
+        for (count i = 0; i < G.numberOfNodes() - n; ++i) { // we have to sample distinct nodes
+            node v = GraphTools::randomNode(G);
+            while (!alreadySelected[v]) {
+                v = GraphTools::randomNode(G);
+            }
+            alreadySelected[v] = false;
+        }
+
+        for (const auto sample : G.nodeRange()) {
+            if (alreadySelected[sample]) {
+                selectedNodes.push_back(sample);
```

Comment:
Break if `selectedNodes.size() == n`

Reply:
Does this make sense? The for loop ends already when you inset the n-th node since that would be all nodes the node range.

Reply:
Does this make sense? if selectedNodes were to reach `n` would mean that they would be caught by the first case, since this the for the case that there are more `n/2` samples but less than `n`, so this break condition would never be reached.

Also for loop ends already when you check the n-th node since that would be all nodes in the node range.